logtest: Add Recorder (#5134)
open-telemetry/opentelemetry-go: CHANGELOG.md

- Add `otel.scope.name` and `otel.scope.version` tags to spans exported by `go.opentelemetry.io/otel/exporters/zipkin`. (#5108)
- Add support for `AddLink` to `go.opentelemetry.io/otel/bridge/opencensus`. (#5116)
- Add `String` method to `Value` and `KeyValue` in `go.opentelemetry.io/otel/log`. (#5117)

> - `InMemory` log recorder implementation to help with testing integrations. (#5134)

Reviewer: ```suggestion
- Add `Recorder` in `go.opentelemetry.io/otel/log/logtest` to facilitate testing the log bridge implementations. (#5134)
```
